scripts/ci/check_compliance.py

```diff
@@ -256,6 +256,28 @@ def get_modules(self, modules_file):
         except subprocess.CalledProcessError as ex:
             self.error(ex.output)
 
+    def write_kconfig_soc(self):
+        """
+        Write KConfig soc files to be sourced during Kconfig parsing
+
+        """
+
+        soc_defconfig_file = os.path.join(tempfile.gettempdir(), "Kconfig.soc.defconfig")
```

Comment:
Are you sure about this? There's only one unique file for this in the entire CI build, and it's /tmp/Kconfig.soc.defconfig? I don't know this code, but this seems like a concurrency problem waiting to happen. What am I missing here?

Reply:
That could be, i'm not to familiar with the new compliance test, to now if that is indeed a risk.
But this code simply follows the existing pattern in the compliance test:
https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/ci/check_compliance.py#L296
```
        # For multi repo support
        self.get_modules(os.path.join(tempfile.gettempdir(), "Kconfig.modules"))
```

However, following an already used pattern elsewhere in this file, is no guarantee that it is correct.
@nashif Could there be a risk of concurrency problem just waiting to happen ?

Reply:
>  @nashif Could there be a risk of concurrency problem just waiting to happen ?

I guess we'll find it the hard way if so. Since this PR is just keeping things consistent with how this file works, I'm assuming it's OK.